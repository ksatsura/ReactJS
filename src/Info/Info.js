//imports
import React from 'react';
import FilmName from '../FilmName/FilmName';
import Genre from '../Genre/Genre';
import ReleaseDate from '../ReleaseDate/ReleaseDate';
import '../style.css';

export const Info = (props) => {

    const { asset } = props;

    return (
        <div className='asset-title'>
            <div className='first-title'>
                <FilmName title={asset.title} />
                <ReleaseDate date={asset.release_date} />
            </div>
            <Genre genres={asset.genres} />
        </div>
    );
};

export default Info;