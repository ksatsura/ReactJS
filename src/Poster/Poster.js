//imports
import React from 'react';
import '../style.css';

const Poster = (props) => {
    
    return <img src={props.url} alt='Poster' height='400px' width='285p' />;
};

export default Poster;