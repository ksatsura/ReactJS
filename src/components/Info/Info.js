//imports
import React from 'react';
import FilmName from '../FilmName/FilmName';
import Genre from '../Genre/Genre';
import ReleaseDate from '../ReleaseDate/ReleaseDate';
import '../../style.css';

export const Info = (props) => {

    const { film } = props;

    return (
        <div className='film-title'>
            <div className='first-title'>
                <FilmName title={film.title} />
                <ReleaseDate date={film.release_date} />
            </div>
            <Genre genres={film.genres} />
        </div>
    );
};

export default Info;