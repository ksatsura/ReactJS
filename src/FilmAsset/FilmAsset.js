//imports
import React from 'react';
import Poster from '../Poster/Poster';
import Info from '../Info/Info';
import { connect } from 'react-redux';
import { fetchPostIfNeeded } from '../actions/assetActions';
import '../style.css';

export const FilmAsset = (props) => {

    const {asset, handleClick} = props;

    return (
        <div id={asset.id} className='film-asset' onClick={(e) => handleClick(e) }>
            <Poster url={asset.poster_path}  />
            <Info asset={asset} />
        </div>
    );
};

const mapDispatchToProps = dispatch => {
    return {
        handleClick: (e) => { dispatch(fetchPostIfNeeded(e.currentTarget.id)); },
    };
};

export default connect(null, mapDispatchToProps)(FilmAsset);