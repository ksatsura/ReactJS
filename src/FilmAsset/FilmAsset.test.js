import React from 'react';
import  { FilmAsset } from './FilmAsset';
import Poster from '../Poster/Poster';
import Info from '../Info/Info';
import { shallow } from 'enzyme';

describe('FilmAsset.js', () => {
    
    const props = {
        asset: {
            poster_path: ''
        },
        handleClick: jest.fn(),
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<FilmAsset {...props} />);

            expect(component).toMatchSnapshot();
            expect(component.find(Info).length).toEqual(1);
            expect(component.find(Poster).length).toEqual(1);
        });

        it('renders for mobile', () => {
            const component = shallow(<FilmAsset {...props} isMobile />);

            expect(component).toMatchSnapshot();
            expect(component.find(Info).length).toEqual(1);
            expect(component.find(Poster).length).toEqual(1);
        });
    });

    it('test film asset click event', () => {
        const component = shallow(<FilmAsset {...props} />);

        component.find('.film-asset').simulate('click');
        expect(props.handleClick.mock.calls.length).toEqual(1);
    });
});


