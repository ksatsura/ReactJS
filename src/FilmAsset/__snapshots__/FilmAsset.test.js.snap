// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmAsset.js renders renders for desktop 1`] = `ShallowWrapper {}`;

exports[`FilmAsset.js renders renders for mobile 1`] = `ShallowWrapper {}`;
